Xero.Api/Infrastructure/RateLimiter/RateLimiter.async.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Xero.Api.Infrastructure.RateLimiter
+{
+    partial class RateLimiter : IAsyncRateLimiter
+    {
+        public Task WaitUntilLimitAsync(CancellationToken cancellation)

```

Comment:
@wesj1s copy the sync version and make this method SleepAsync then use await SleepAsync instead of Thread.Sleep in async version